### ISSUE TITLE
Array Index Lens

### ISSUE THREAD

**Issue by utatti** [utatti/lens.ts#1]
``` typescript
type Person = {
    name: string,
    age: number,
    accounts: string[],
};

const azusa: Person = {
    name: 'Azusa',
    age: 15,
    accounts: ['@azusa', '@nakano'],
};

// property lenses are created dynamically with Proxy
const secondAccountL = id<Person>().accounts[1];

console.log(getl(secondAccountL)(azusa)); // -> '@nakano'
```

The code above works, but the type of `secondAccountL` becomes `any`. I'm not sure of how to make it type-safe on array properties.

**Comment by utatti** [comment]
I think now it's possible with 2.8 conditional types. Will try.

**Comment by utatti** [comment]
Resolved with 0.2.0

- [`.i(index: number)`](https://github.com/utatti/lens.ts#iindex-number)
